test/js/bun/http/serve.test.ts

```diff
@@ -1856,3 +1856,36 @@ it("we should always send date", async () => {
     expect(res.headers.has("Date")).toBeTrue();
   }
 });
+
+it("should allow use of custom timeout", async () => {
+  using server = Bun.serve({
+    port: 0,
+    idleTimeout: 4, // uws precision is in seconds, and lower than 4 seconds is not reliable its timer is not that accurate
+    async fetch(req) {
+      const url = new URL(req.url);
+      return new Response(
+        new ReadableStream({
+          async pull(controller) {
+            controller.enqueue("Hello,");
+            if (url.pathname === "/timeout") {
+              await Bun.sleep(5000);
+            } else {
+              await Bun.sleep(10);
+            }
+            controller.enqueue(" World!");
+
+            controller.close();
+          },
+        }),
+        { headers: { "Content-Type": "text/plain" } },
+      );
+    },
+  });
+  const res = await fetch(new URL("/timeout", server.url.origin));
```

Comment:
Can we run both of these `fetch` calls at the same time? Or is the goal to test with keep alive?

Reply:
we can test in parallel fixed in https://github.com/oven-sh/bun/pull/13453/commits/284d24187633ec379f1cc7b492018fd071dad636